src/main/java/io/cryostat/credentials/Credentials.java

```diff
@@ -105,6 +109,7 @@ private void createFromFile(java.nio.file.Path path) {
                                     .count()
                             != 0;
             if (exists) {
+                logger.trace("Credential exists");
```

Comment:
If the credential already existed, then maybe the match expression we just created should be deleted?

There should be a nice way to do this using transaction rollbacks. https://quarkus.io/guides/transaction

Reply:
There's a better solution I think that avoids creating another matchExpression in this case, we can query it with MatchExpression.find and check if that one comes back true and the credentials query for username/pass returns true. Updated the PR.

Reply:
Are you sure this is a complete solution? The MatchExpression entity does not have a uniqueness constraint on the `script` field and the field is not part of the entity's primary key for identifying it. So checking if a Match Expression entity already exists that has the same `script` contents does not seem like it means that persisting this Credential instance will succeed. I don't think there is any implementation currently that looks for identical scripts within expressions and reuses the expression instances, and I'm not sure if that's the road we want to take either.

Reply:
Since there is now a `credential.matchExpression.persist()` call above, I also think that this `.find() != 0` query is always going to turn up `true`.

Reply:
> Since there is now a credential.matchExpression.persist() call above, I also think that this .find() != 0 query is always going to turn up true.

I think this is still the case - should the `.find()` query be done before the `.persist()`?

Reply:
The MatchExpression can't be referenced in a query without being persisted first, that was the original bug and was causing the TransientObjectException. This allows for duplicates however since persisting it creates a new ID, so this query with the current data format will always return true. 

We can probably remove it in lieu of the comment about reworking how MatchExpressions are stored (#912) since it's not doing anything. What do you think?

Reply:
Ah right, yes I see what you mean. Then let's go ahead and remove that part. There are other mitigations we can and should apply at a higher level, ex. in the Helm chart or Operator Deployments we should try to avoid mounting Secrets/ConfigMaps in a way where the same declarative config file can get loaded twice.

Maybe in this or a follow-up PR it would also be worth exploring that symlink deduplication idea that we discussed. That doesn't fix the underlying ambiguity in the data model, but it does at least prevent the declarative config startup from tripping over the underlying bug.

Reply:
https://github.com/cryostatio/cryostat/issues/916